### PR TITLE
8254075: Shenandoah: Remove ShenandoahCodeRootsStyle diagnostic flag and related test

### PR DESCRIPTION
**Pull request by zhengyu123**
Shenandoah introduced fast code roots iteration (ShenandoahCodeRootsStyle = 2), also kept 2 additional iteration mechanism (ShenandoahCodeRootsStyle = 0 and 1), mainly for verification purpose.

Today, Shenandoah normally iterates code roots concurrently using style 2, while style 0 and 1 require safepoints, that makes style 0 and 1 verification less useful. Besides, Shenandoah has much improved verifier, more suitable for the purpose.

Let's remove this diagnostic flag and related test.

Test:

- [x] hotspot_gc_shenandoah

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8254075](https://bugs.openjdk.java.net/browse/JDK-8254075): Shenandoah: Remove ShenandoahCodeRootsStyle diagnostic flag and related test

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/524/head:pull/524`
`$ git checkout pull/524`

### CODE REVIEW

File: src/hotspot/share/gc/shenandoah/shenandoahConcurrentRoots.cpp [hunk 1]
```diff
@@ -37,12 +37,7 @@ bool ShenandoahConcurrentRoots::should_do_concurrent_roots() {
 }
 
 bool ShenandoahConcurrentRoots::can_do_concurrent_class_unloading() {
-#if (defined(X86) || defined(AARCH64))
-  return ShenandoahCodeRootsStyle == 2 &&
-         ClassUnloading;
-#else
-  return false;
-#endif
+  return true;
```
Comment (shipilev) [inline src/hotspot/share/gc/shenandoah/shenandoahConcurrentRoots.cpp:40]: Hold on, this used to check `ClassUnloading`, should it continue to check it?

**Review by shipilev — CHANGES_REQUESTED** [review]
Looks fine to me, except `ClassUnloading` handling in one place.

File: src/hotspot/share/gc/shenandoah/shenandoahConcurrentRoots.cpp (hunk 1 above)
Comment (zhengyu123) [reply src/hotspot/share/gc/shenandoah/shenandoahConcurrentRoots.cpp:40]: No. I actually puzzled why checked it here, our concurrent roots scanner can handle both scenarios.
Maybe a leftover from concurrent roots to concurrent class unloading transition?

Comment (shipilev) [reply src/hotspot/share/gc/shenandoah/shenandoahConcurrentRoots.cpp:40]: It is maybe something that disables concurrent class unloading when `-ClassUnloading` is specified? Although it should have been handled in `shenandoahArguments.cpp`. Still, I'd prefer to leave the behavior exact, and then remove the `ClassUnloading` reference in a separate PR.

Comment (zhengyu123) [reply src/hotspot/share/gc/shenandoah/shenandoahConcurrentRoots.cpp]: Make sense, updated.

**Review by shipilev — APPROVED** [review]

**Pull request closed without merging**